samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemAddListener.java

```diff
@@ -1148,6 +1148,23 @@ else if (itemauthor.getTarget() != null && (itemauthor.getTarget().equals("samba
 	  }
   }
 
+ /** 
+  * Helper method to get a list of ValidAnswers for a matching question from a MatchItemBeanList
+  * @param matchItemBeanList 
+  **/
+ private List<MatchItemBean> getValidMIBAnswersList (List<MatchItemBean> matchItemBeanList) {
+	  // Create a list of valid answers to loop through.  Ignore answers that are distractors
+	  // or are controlled by another MatchItemBean
+	  List<MatchItemBean> validAnswers = new ArrayList<MatchItemBean>();
+	  Iterator<MatchItemBean>validAnswerIter = matchItemBeanList.iterator();
+	  for (MatchItemBean validAnswer: matchItemBeanList) {
+		  if (MatchItemBean.CONTROLLING_SEQUENCE_DEFAULT.equals(validAnswer.getControllingSequence())) {
+			  validAnswers.add(validAnswer);
+		  }
+	  }
+	  return validAnswers;
```

Comment:
This can be reduced to:
```
	  List<MatchItemBean> validAnswers = matchItemBeanList.stream()
			  .filter(answer -> MatchItemBean.CONTROLLING_SEQUENCE_DEFAULT.equals(answer.getControllingSequence()))
			  .collect(Collectors.toList());
```

Reply:
That seems goo. I'm still a little traditional and I thought this was a minor improvement on what was there. :) I don't think I'll have a chance to commit and compile this until tomorrow night so if you want to commit and push it, go for it!